processing/src/main/java/org/apache/druid/segment/column/TypeStrategy.java

```diff
@@ -216,4 +216,9 @@ default boolean equals(T a, T b)
   {
     throw DruidException.defensive("Not implemented. Check groupable() first");
   }
+
+  default Class<?> complexDimensionType()
```

Comment:
nit: maybe `getComplexClass` or the classic `getClazz` would be a better name since it probably isn't a different type when used as a dimension (and we have the separate `groupable` to determine if it can be used as a dimension or not)